File: src/main/service/CarService.java
package main.service;

import main.Reference;
import main.dao.CarDAO;
import main.model.Car;

import java.util.List;

public class CarService
{
    private CarDAO carDAO;

    public CarService() {
        this.carDAO = new CarDAO();
    }

//    public boolean addcar(Car car)
//    {
//        return this.carDAO.addcar(car);
//    }

    public boolean addcar(Car car, Reference<List<Car>> carlist)
    {
        boolean result = this.carDAO.addcar(car);
        carlist.get().add(car);
        return result;
    }

    public boolean removecar(int id_car, Reference<List<Car>> carlist){
        boolean result =  this.carDAO.removecar(id_car);

        for( Car itemcar : carlist.get()) {
            if(itemcar.getId_car() == id_car)
            {
                carlist.get().remove(itemcar);
                break;
            }
        }
        return result;
    }



    public List<Car> getAllCarModel(String username,boolean force) {
        return this.carDAO.getAllCarModel(username,force);
    }

    public Car getCar(int id_car,boolean force){
        return this.carDAO.getCar(id_car,force);
    }
}

File: PCarpet/Service/CarService.cs
using PCarpet.DTO;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace PCarpet.Service
{
    public class CarService
    {

        public void addCar(CarDTO carDTO)
        {
            using (pcarpetEntities context = new pcarpetEntities())
            {
                context.car.Add(new car(carDTO));
                context.SaveChanges();
            }
                
        }


        public void removeCar(int id)
        {
            using (pcarpetEntities context = new pcarpetEntities())
            {
                car car = getCar(id);
                context.car.Attach(car);
                context.car.Remove(car);
                context.SaveChanges();
            }
        }


        public List<CarDTO> getAllCarDTO(string username)
        {
            using (pcarpetEntities context = new pcarpetEntities())
            {
                List<car> cars = context.car.Where(car => car.username.Equals(username)).ToList();
                List<CarDTO> carsDTO = new List<CarDTO>();
                foreach (car car in cars)
                {
                    carsDTO.Add( car.toCarDTO(car) );
                }
                return carsDTO;
            }              
        }



        //metodi interni
        public car getCar(int id_car)
        {
            using (pcarpetEntities context = new pcarpetEntities())
            {
                return context.car.FirstOrDefault(c => c.id.Equals(id_car));
            }
        }

        public List<car> getAllCar(string username)
        {
            using (pcarpetEntities context = new pcarpetEntities())
            {
                return context.car.Where(car => car.username.Equals(username)).ToList();
            }
        }
    }
}